src/cursor/cur_table.c

```diff
@@ -849,17 +849,19 @@ __curtable_bound(WT_CURSOR *cursor, const char *config)
     JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
 
     /* Save the current state of the bounds in case we fail to apply the new state. */
-    __wt_cursor_bounds_save(primary, &saved_bounds);
```

Comment:
This should be a bool that we save so we don't trick ourselves into restoring state we didn't have previously. As we would make `WT_CURSOR_BOUNDS_SET` true by applying bounds further down.